docker/cytoscape/src/connect_cytoscape.py

```diff
@@ -1,257 +1,333 @@
+            res=r.json()
+            if "errors" in res.keys():
+                if len(res["errors"]) > 0:
+                    raise ValueError(res["errors"][0])
+        except JSONDecodeError:
+            if not r.text:
+                res = {}
+            else:
+                raise
+
+    elif (method=="HTML") or (method == "H") or (method=="HELP"):
+        URL = baseurl
+        if verbose:
+            print("'"+URL+"'")
+            sys.stdout.flush()
+        response = requests.get(URL, params=PARAMS)
+
+        res = response.text.split("\n")
+        def clean(x):
+            r=x.split("</p>")[0].split(">")[-1]
+            return r
+        res="\n".join([ clean(x) for x in res ])
+
+    res=handle_status_codes(res,verbose=verbose)
+
+    return res
+
+def _status(cyrest_client, host, port, verbose=False):
+    """
+    Checks the status of your CyREST server.
+    """
+    _url = 'http://' + host + ':' + str(port) + '/' + 'v1' + '/'
+    try:
+        response=api(url=_url, method="GET", verbose=verbose)
+        print('CyREST online!')
+        return response
+    except Exception as e:
+        print('Connecting to Cytoscape ...', flush=True)
+
+##########################################################################################################
+##########################################################################################################
+       
+CYTOSCAPE_DATATYPES = 's,s,b,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl,dl'
+
+def get_parser() -> argparse.ArgumentParser:
+    '''
+    :return: an argparse ArgumentParser object for parsing command
+        line parameters
+    '''
+    parser = argparse.ArgumentParser(
+        description='Run Cytoscape pipeline.')
+
+    parser.add_argument('--save',
+        help='cytoscape session file')
+    parser.add_argument('--style', help='style file')
+    parser.add_argument('--annot', help='annotations file')
+    parser.add_argument('--network', help='network file')
+    parser.add_argument('--datatypes', help='data types for columns')
+
+    return parser
+
+def parse_arguments():
+    '''
+    Initialize a parser and use it to parse the command line arguments
+    :return: parsed dictionary of command line arguments
+    '''
+    parser = get_parser()
+    opts = parser.parse_args()
+
+    return opts
+
+class CytoscapeClient:
+
+    def __init__(self, save_file, style_file, annot_file, network_file, data_types) -> None:
+        self.save_file = save_file
+        self.style_file = style_file
+        self.annot_file = annot_file
+        self.network_file = network_file
+        self.data_types = data_types
+
+    def __visualize_outputs(self, cyclient):
+        _annot_file = os.path.abspath(self.annot_file)
+        _style_file = os.path.abspath(self.style_file)
+
+        cy = CyRestClient()
+        cy.session.delete()
+        cy.network.create_from(self.network_file)
+        time.sleep(2)
+        style = cyclient.vizmap.load_file(_style_file)

```

Comment:
Do we need to be more flexible in case there isn't a style file or annotations file?  Or do we expect that because this is only used with TPS, we'll always have those inputs?